src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewTemplate.cs

```diff
@@ -53,15 +53,14 @@ internal static async Task RenderAsync(RazorPage<dynamic> page)
 
             try
             {
-                var content = new ViewBufferTextWriterContent();
+                using var content = new ViewBufferTextWriterContent();
                 await context.EnterScopeAsync(page.ViewContext, (object)page.Model);
                 await template.FluidTemplate.RenderAsync(content, htmlEncoder, context);
-                
-                // Use ViewBufferTextWriter.Write(object) from ASP.NET directly since it will use a special code path
-                // for IHtmlContent. This prevent the TextWriter methods from copying the content from our buffer 
-                // if we did content.WriteTo(page.Output)
-                
-                page.Output.Write(content);
+
+                // Use ViewBufferTextWriter.WriteAsync(string value) from ASP.NET directly since so it will write content from
+                // pooled string builder and allow us to return it to pool by having the using directive above
+
+                await page.Output.WriteAsync(content.ToString());
```

Comment:
We intentionally pass the `IHtmlContent` instance such that it calls `WriteTo`, which will release the resources.